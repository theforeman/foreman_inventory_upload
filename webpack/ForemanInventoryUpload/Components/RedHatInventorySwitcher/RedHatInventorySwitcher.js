import React, { useState } from 'react';
import { translate as __ } from 'foremanReact/common/I18n';
import ShortSwitcherPF4 from '../../../common/Switcher/ShortSwitcherPF4';

const RedHatInventorySwitcher = () => {
  const [redHatInventoryEnabled, setRedHatInventoryEnabled] = useState(false);
  return (
    <ShortSwitcherPF4
      id="rh-inventory-settings-switcher"
      label={__('Red Hat Inventory')}
      isChecked={redHatInventoryEnabled}
      onChange={newIsChecked => setRedHatInventoryEnabled(newIsChecked)}
    />
  );
};

export default RedHatInventorySwitcher;
