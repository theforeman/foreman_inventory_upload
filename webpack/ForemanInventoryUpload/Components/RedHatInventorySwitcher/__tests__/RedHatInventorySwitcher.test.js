import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import RedHatInventorySwitcher from '../RedHatInventorySwitcher';

const fixtures = {
  render: {},
};

describe('RedHatInventorySwitcher', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(RedHatInventorySwitcher, fixtures));
});
