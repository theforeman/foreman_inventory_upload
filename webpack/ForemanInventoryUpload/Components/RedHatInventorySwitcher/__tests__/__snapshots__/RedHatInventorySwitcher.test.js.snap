// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedHatInventorySwitcher rendering render 1`] = `
<ShortSwitcherPF4
  id="rh-inventory-settings-switcher"
  isChecked={false}
  label="Red Hat Inventory"
  onChange={[Function]}
  tooltip={null}
/>
`;
